com.unity.multiplayer.mlapi/Runtime/Core/NetworkingManager.cs

```diff
@@ -92,17 +92,11 @@ public class NetworkingManager : MonoBehaviour, INetworkUpdateSystem
         /// The log level to use
         /// </summary>
         [HideInInspector]
-        public LogLevel LogLevel = LogLevel.Normal;
-        /// <summary>
-        /// The singleton instance of the NetworkingManager
-        /// </summary>
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        [Obsolete("Use Singleton instead", false)]
-        public static NetworkingManager singleton => Singleton;
-        /// <summary>
-        /// The singleton instance of the NetworkingManager
-        /// </summary>
-        public static NetworkingManager Singleton { get; private set; }
```

Comment:
I think we should get rid of this singleton boss but this PR looks like a mess to me.
_(I know this is not a very constructive feedback, sorry)._

There are lots of things going on here:
- `NetManager`/`NetworkManager`, `NetObject` and other naming inconsistencies
- Assumptions like: ![](https://user-images.githubusercontent.com/12574651/109304510-4c612700-7834-11eb-8f86-119d3f387553.png)
- Controversial changes like pinning `PooledBitReader`, `InternalMessageSender` etc to `NetworkManager`: `using (PooledBitReader reader = NetManager.PooledBitReaders.GetReader(stream))`
- other fundamental behavioral changes which could be counted as high risks

Overall, I believe we need to redo/rework singleton removal again and think about design changes more carefully.

However, if I should consider this PR only as a draft that's never going to be merged into `develop`, then I'm fine :)

> This is a draft PR so we can have a place to discuss the singleton removal

I wanted to double-check the statement above.